apps/mark-scan/backend/src/custom-paper-handler/application_driver.ts

```diff
@@ -0,0 +1,225 @@
+  for (const customChunkedBitmap of customChunkedBitmaps) {
+    if (customChunkedBitmap.empty) {
+      dotsSkipped += VERTICAL_DOTS_IN_CHUNK;
+    } else {
+      if (dotsSkipped) {
+        await driver.setRelativeVerticalPrintPosition(dotsSkipped * 2); // assuming default vertical units, 1 / 408 units
+        dotsSkipped = 0;
+      }
+      await driver.printChunk(customChunkedBitmap);
+    }
+  }
+  debug('Completed printing %d chunks total', customChunkedBitmaps.length);
+}
+
+export async function scanAndSave(
+  driver: PaperHandlerDriver
+): Promise<SheetOf<string>> {
+  debug('+scanAndSave');
+  const dir = '/tmp';
+  const dateString = new Date().toISOString();
+  const pathOutFront = join(dir, `ballot-statemachine-${dateString}-A.jpg`);
+  // We can only print to one side from the thermal printer, but the interpret flow expects
+  // a SheetOf 2 pages. Use an image of a blank sheet for the 2nd page.
+  const blankSheetFixturePath = join(__dirname, 'fixtures', 'blank-sheet.jpg');
+  const status = await driver.getPaperHandlerStatus();
+  // Scan can happen from loaded or parked state. If the paper is not loaded or parked
+  // it means the voter may have taken the paper out of the infeed
+  if (!isPaperAnywhere(status)) {
+    throw new Error('Paper has been removed');
+  }
+
+  await driver.scanAndSave(pathOutFront);
+  debug('Scan successful');
+  return [pathOutFront, blankSheetFixturePath];
+}
+
+function combinePageInterpretationsForSheet(
+  pages: SheetOf<PageInterpretationWithFiles>
+): SheetInterpretation {
+  const [front, back] = pages;
+  const frontType = front.interpretation.type;
+  const backType = back.interpretation.type;
+
+  // mark-scan only support interpretation of single-sided BMD ballots
+  if (
+    (frontType === 'InterpretedBmdPage' && backType === 'BlankPage') ||
+    (backType === 'InterpretedBmdPage' && frontType === 'BlankPage')
+  ) {
+    return { type: 'ValidSheet' };
+  }
+
+  debug(
+    'Unexpected page interpretation: %s front type and %s back type',
+    frontType,
+    backType
+  );
+
+  return {
+    type: 'InvalidSheet',
+    reason: 'unknown',
+  };
+}
+
+export async function interpretScannedBallots(
+  electionDefinition: ElectionDefinition,
+  precinctSelection: PrecinctSelection,
+  testMode: boolean,
+  sheetOfImagePaths: SheetOf<string>,
+  sheetId: string
+): Promise<SheetInterpretationWithPages> {
+  const pageInterpretations = await interpretSheetAndSaveImages(
+    {
+      electionDefinition,
+      precinctSelection,
+      testMode,
+    },
+    sheetOfImagePaths,
+    sheetId,
+    BALLOT_IMAGES_PATH
+  );
+
+  const combinedInterpretation: SheetInterpretationWithPages = {
+    ...combinePageInterpretationsForSheet(pageInterpretations),
+    pages: pageInterpretations,
+  };
+
+  // Use JSON.stringify instead of string interpolation because the latter
+  // only prints one level deep
+  debug(`interpretation: ${JSON.stringify(combinedInterpretation, null, 2)}`);
```

Comment:
Does this do what you want?
```suggestion
  debug('interpretation: %O', combinedInterpretation);
```

Reply:
It prints the first level but level 2 evaluates to `[object Object]` :(